src/main/java/org/generationcp/middleware/dao/ims/TransactionDAO.java

```diff
@@ -737,33 +758,15 @@ public long countSearchTransactions(final TransactionsSearchDto transactionsSear
 
 	private Constructor<TransactionDto> getTransactionDtoConstructor() {
 		try {
-			return TransactionDto.class.getConstructor(Integer.class, String.class, String.class,  Double.class,
-			 String.class, Date.class, Integer.class, Integer.class, String.class, String.class, Integer.class, String.class, String.class);
+			return TransactionDto.class.getConstructor(Integer.class, String.class, String.class, Double.class,
+				String.class, Date.class, Integer.class, Integer.class, String.class, String.class, Integer.class, String.class,
+				String.class, String.class);
 		} catch (final NoSuchMethodException ex) {
 			throw new RuntimeException(ex);
 		}
 	}
 
-	public Transaction saveTransaction(final TransactionDto transactionDto) {
-		final Lot lot = new Lot();
-		lot.setId(transactionDto.getLot().getLotId());
-		final Transaction transaction = new Transaction();
-		if (TransactionType.DEPOSIT.getValue().equalsIgnoreCase(transactionDto.getTransactionType()) || TransactionType.WITHDRAWAL
-			.getValue().equalsIgnoreCase(transactionDto.getTransactionType())) {
-			transaction.setStatus(TransactionStatus.COMMITTED.getIntValue());
-		} else {
-			transaction.setStatus(TransactionStatus.ANTICIPATED.getIntValue());
-		}
-		transaction.setLot(lot);
-		transaction.setPersonId(Integer.valueOf(transactionDto.getCreatedByUsername()));
-		transaction.setUserId(Integer.valueOf(transactionDto.getCreatedByUsername()));
-		transaction.setTransactionDate(new Date());
-		transaction.setQuantity(transactionDto.getAmount());
-		transaction.setPreviousAmount(0D);
-		//FIXME Commitment date in some cases is not 0. For Deposits is always zero, but for other types it will be the current date
-		transaction.setCommitmentDate(0);
-		transaction.setComments(transactionDto.getNotes());
-
+	public Transaction saveTransaction(final Transaction transaction) {
```

Comment:
Not clear why you need to expose the saveTransaction instead of  using saveOrUpdate directly.

Reply:
Solved